WordPress/wordpress-playground
# Blueprints: Remove the applyWordPressPatches step, enable the Site Health Plugin

Prior to this PR, Playground would apply a series of patches to the WordPress source code to make it compatible with the PHP-Wasm runtime.

This PR removes most of those patches and the entire `applyWordPressPatches` step.

## List of changes

* The Site Health plugin is now enabled by default. Playground's PHP.wasm now supports enough PHP features to enable that.
* The `wp_new_blog_notification` function is no longer overridden as one shipped by WordPress works.
* Network transports are now a part of the Playground mu-plugin and are not created conditionally. This solves #977.
* Playground mu-plugin lives in the playground-remote package, not in the Blueprints package. The plugin is only used by the Web Worker where Playground is set up, so let's bundle those together.
* The `PHPRequestHandler` now returns a 502 status code when a request is received while another request is still being handled. Previously it would deadlock as handling a request requires an exclusive lock on the PHP instance. This prevents the Site Health plugin from triggering a deadlock by requesting the WordPress site from inside the PHP code.

## Testing instructions

* Confirm the CI checks pass.
* Customize the WordPress site in Playground, export it, restart Playground, import the file, confirm it worked.
* Try out the WordPress PR previewer and confirm it still works.

cc @bgrgicak
